lib/spack/spack/build_environment.py

```diff
@@ -195,6 +195,15 @@ def set_compiler_environment_variables(pkg, env):
     env.set('SPACK_F77_RPATH_ARG', compiler.f77_rpath_arg)
     env.set('SPACK_FC_RPATH_ARG',  compiler.fc_rpath_arg)
 
+    # Set linker behavior to use rpaths
+    # necessary for ubuntu 17+
+    # HOTFIX: TODO: Something better
+    rpath_arg = ''
+    pkg_os = pkg.spec.architecture.platform_os
+    if pkg_os.startswith('ubuntu') and int(pkg_os[6:8]) >= 17:
```

Comment:
@becker33 

I think this could be safer if slightly modified:

`if pkg_os.startswith('ubuntu') and pkg_os[6:8] and int(pkg_os[6:8]) >= 17`

basically just make sure that substring exists so it doesn't try `int('')` and get an error

Other than that this looks good for the time being